### ISSUE TITLE
Rails/RelativeDateConstant complains about relative dates in procs

### ISSUE THREAD

**Issue by buehmann** [bbatsov/rubocop#4196]
We have some code like
```ruby
CONST = -> { 1.year.ago }

CONST.call
```
By using the lambda we make sure to re-evaluate `1.year.ago`when needed.

## Expected behavior

`Rails/RelativeDateConstant` should not complain about this.

## Actual behavior

```
Offenses:

bug.rb:1:1: C: Rails/RelativeDateConstant: Do not assign ago to constants as it will be evaluated only once.
CONST = -> { 1.year.ago }
^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

Above code as `bug.rb`; then `rubocop --rails bug.rb -D`

## RuboCop version

```
$ rubocop -V
0.48.0 (using Parser 2.4.0.0, running on ruby 2.3.1 x86_64-darwin15)
```
